### PR TITLE
refactor: Move accept content type code from App into ApiRequest

### PR DESCRIPTION
**Pull request by monacoremo**
This is a smaller intermediate step to carving out a small API to `Request`. This moves the code that finds an acceptable response content type into ApiRequest.

### CODE REVIEW

File: src/PostgREST/Request/ApiRequest.hs [hunk 1]
```diff
@@ -158,22 +159,25 @@ data ApiRequest = ApiRequest {
...
   , iProfile              :: Maybe Schema                     -- ^ The request profile for enabling use of multiple schemas. Follows the spec in hhttps://www.w3.org/TR/dx-prof-conneg/ttps://www.w3.org/TR/dx-prof-conneg/.
   , iSchema               :: Schema                           -- ^ The request schema. Can vary depending on iProfile.
+  , iAcceptContentType    :: ContentType
   }
 
 -- | Examines HTTP request and translates it into user intent.
-userApiRequest :: NonEmpty Schema -> Maybe Text -> DbStructure -> Request -> RequestBody -> Either ApiRequestError ApiRequest
-userApiRequest confSchemas rootSpec dbStructure req reqBody
+userApiRequest :: AppConfig -> DbStructure -> Request -> RequestBody -> Either ApiRequestError ApiRequest
+userApiRequest conf dbStructure req reqBody
```
Comment (wolfgangwalther) [inline src/PostgREST/Request/ApiRequest.hs:167]:
```suggestion
userApiRequest AppConfig{..} dbStructure req reqBody
```
The where list is long enough and `confSchemas` is used immediately on the next line - so using `configDbSchemas` is easier to follow.

**Review by wolfgangwalther — APPROVED** [review]

File: src/PostgREST/Request/ApiRequest.hs [hunk 2]
```diff
@@ -158,22 +159,25 @@ data ApiRequest = ApiRequest {
...
-  | otherwise = Right ApiRequest {
+  | otherwise = do
+     acceptContentType <- findAcceptContentType conf action target accepts
+     return ApiRequest {
       iAction = action
       , iTarget = target
       , iRange = ranges
       , iTopLevelRange = topLevelRange
-      , iAccepts = maybe [CTAny] (map ContentType.decodeContentType . parseHttpAccept) $ lookupHeader "accept"
+      , iAccepts = accepts
```
Comment (wolfgangwalther) [inline src/PostgREST/Request/ApiRequest.hs:180]: I think this field is not needed at all anymore, because `accepts` is used right in here.

**Comment by wolfgangwalther** [comment]
Cirrus is failing with:
```
src/PostgREST/App.hs:163:34: error: [-Wunused-record-wildcards, -Werror=unused-record-wildcards]
    No variables bound in the record wildcard match are used
      Possible fix: omit the ‘..’
    |
163 | postgrestResponse conf@AppConfig{..} maybeDbStructure pool time req = do
```

I wonder why none of our other builds fail here? I would expect that this should fail everywhere.

File: src/PostgREST/Request/ApiRequest.hs (hunk 2 above)
Comment (monacoremo) [reply src/PostgREST/Request/ApiRequest.hs:180]: True, was not needed anymore!

**Comment by monacoremo** [comment]
> I wonder why none of our other builds fail here? I would expect that this should fail everywhere.

```
Build profile: -w ghc-8.10.4 -O1
```

Apparently it builds with GHC 8.10 that added this warning. Wouldn't that require a different stack resolver though?..

**Pull request merged**